platform/ui/src/components/content/viewportDownloadForm/ViewportDownloadForm.js

```diff
@@ -62,125 +76,231 @@ const ViewportDownloadForm = ({
     height: defaultSize,
   });
 
-  // Cornerstone's `enable/disable`
-  useEffect(() => {
-    enableViewport(viewportElement);
+  const [error, setError] = useState({
+    width: false,
+    height: false,
+    filename: false,
+  });
 
-    return () => {
-      disableViewport(viewportElement);
-    };
-  }, [disableViewport, enableViewport, viewportElement]);
+  const hasError = Object.values(error).includes(true);
 
-  useEffect(() => {
-    const { width, height } = viewportElementDimensions;
-    const validSize = value => (value >= minimumSize ? value : minimumSize);
-    const loadAndUpdateViewports = async () => {
-      await loadImage(activeViewport, viewportElement, width, height);
-      toggleAnnotations(showAnnotations, viewportElement);
-
-      const {
-        dataUrl,
-        width: viewportElementWidth,
-        height: viewportElementHeight,
-      } = await updateViewportPreview(
-        viewportElement,
-        downloadCanvas.ref.current,
-        fileType
-      );
-
-      setViewportPreview(state => ({
-        ...state,
-        src: dataUrl,
-        width: validSize(viewportElementWidth),
-        height: validSize(viewportElementHeight),
-      }));
-    };
+  const refreshViewport = useRef(null);
 
-    loadAndUpdateViewports();
-  }, [
-    activeViewport,
-    viewportElement,
-    showAnnotations,
-    loadImage,
-    toggleAnnotations,
-    updateViewportPreview,
-    fileType,
-    downloadCanvas.ref,
-    minimumSize,
-    maximumSize,
-    viewportElementDimensions,
-  ]);
+  const downloadImage = () => {
+    downloadBlob(
+      filename || DEFAULT_FILENAME,
+      fileType,
+      viewportElement,
+      downloadCanvas.ref.current
+    );
+  };
 
   /**
-   * @param {object} event - Input change event
-   * @param {string} dimension - "height" | "width"
-   */
+  * @param {object} event - Input change event
+  * @param {string} dimension - "height" | "width"
+  */
   const onDimensionsChange = (event, dimension) => {
+    const opositeDimension = dimension === 'height' ? 'width' : 'height';
```

Comment:
A little typo? opposite  👀 (me being picky)

Reply:
Thanks for seeing that . :)